cloud/amazon/ecs_taskdefinition.py

```diff
@@ -120,14 +122,95 @@ def __init__(self, module):
                 module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION environment variables or in boto configuration file")
             self.ecs = boto3_conn(module, conn_type='client', resource='ecs', region=region, endpoint=ec2_url, **aws_connect_kwargs)
         except boto.exception.NoAuthHandlerFound, e:
-            module.fail_json(msg="Can't authorize connection - "+str(e))
+            self.module.fail_json(msg="Can't authorize connection - "+str(e))
 
     def describe_task(self, task_name):
         try:
+            families_response = self.ecs.list_task_definition_families()
+            if task_name not in families_response['families']:
+                return None
+
             response = self.ecs.describe_task_definition(taskDefinition=task_name)
             return response['taskDefinition']
-        except botocore.exceptions.ClientError:
-            return None
+        except botocore.exceptions.ClientError, e:
+            self.module.fail_json(msg="Can't describe task - "+str(e))
+
+    def is_matching_task(self, expected, existing):
+        # prepare containers for comaprison
+        existingContainers = list(existing['containerDefinitions']) if 'containerDefinitions' in existing else []
+        expectedContainers = list(expected['containers']) if 'containers' in expected else []
+
+        for containers in [existingContainers, expectedContainers]:
+            containers.sort(key=lambda k: k['name'])
+            for container in containers:
+                # in case some properties are not defined, we put the default values
+                # and sort lists properties for comaprison
+
+                prop_defaults = {
+                    'disableNetworking': False,
+                    'privileged': False,
+                    'readonlyRootFilesystem': False
+                }
+
+                for prop_name, prop_default in prop_defaults.iteritems():
+                    if prop_name not in container:
+                        container[prop_name] = prop_default
+
+                simple_list_props = [
+                    'links',
+                    'command',
+                    'entryPoint',
+                    'dnsServers',
+                    'dnsSearchDomains',
+                    'dockerSecurityOptions']
+
+                for list_prop in simple_list_props:
+                    if list_prop not in container:
+                        container[list_prop] = []
+                    else:
```

Comment:
I've been hitting a problem with my entryPoint commands.  Basically, in my playbook I have:

```
entryPoint:
            - "bash"
            - "-c"
```

It needs to be in that order, but I think because of the sort here it is getting sorted to:

```
entryPoint:
            - "-c"
            - "bash"
```

Which causes the container not to start (Docker can't find the command "-c")

Suggest changing this line to: